sdk/storage/AzureStorageBlob/Source/StorageBlobClientDelegate.swift

```diff
@@ -27,13 +27,30 @@
 import Foundation
 
 public protocol StorageBlobClientDelegate: AnyObject {
+    /// A blob transfer's state and/or progress have changed.
+    /// - Parameters:
+    ///   - client: The `BlobStorageClient` associated with the transfer.
+    ///   - transfer: The `BlobTransfer` that has changed.
+    ///   - state: The `TransferState` of the blob transfer.
+    ///   - progress: The `TransferProgress` of the blob transfer, if relevant.
     func blobClient(
         _ client: StorageBlobClient,
         didUpdateTransfer transfer: BlobTransfer,
         withState state: TransferState,
-        andProgress progress: Float?
+        andProgress progress: TransferProgress?
```

Comment:
Maybe we can make this non-nullable? If the progress is zero, or 100% that's fine.

Reply:
Done